### DISCUSSION TITLE
有的图片报这个错误，怎么解决？ValueError: cannot select an axis to squeeze out which has size not equal to one

### DISCUSSION THREAD

**Discussion by nissansz** [RapidAI/TableStructureRec#5] [Q&A]
File "F:/pycharm2020.2/RapidStructure-0.0.0/TableStructureRec-main/demo_wired.py", line 31, in <module>
    table_str, elapse = table_rec(img_path)
  File "F:\pycharm2020.2\RapidStructure-0.0.0\TableStructureRec-main\wired_table_rec\main.py", line 47, in __call__
    polygons = self.table_line_rec(img)
  File "F:\pycharm2020.2\RapidStructure-0.0.0\TableStructureRec-main\wired_table_rec\table_line_rec.py", line 69, in __call__
    polygons = merge_adjacent_polys(polygons)
  File "F:\pycharm2020.2\RapidStructure-0.0.0\TableStructureRec-main\wired_table_rec\utils_table_recover.py", line 79, in merge_adjacent_polys
    polygons = combine_two_poly(polygons, idxs)
  File "F:\pycharm2020.2\RapidStructure-0.0.0\TableStructureRec-main\wired_table_rec\utils_table_recover.py", line 88, in combine_two_poly
    idxs = idxs.squeeze(0)
ValueError: cannot select an axis to squeeze out which has size not equal to one

原图
![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/2c93c837-f3c2-4b4a-b28c-1c9e611a420a)

可视化表格框
![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/a94f7fad-b578-4d3d-be4d-f334fad76156)

**Comment by qqeip** [comment]
这些都是 阿里读光的表格识别模型 

考虑到现有表格识别模型较弱，尝试阿里读光下的表格识别模型效果。
[读光-表格结构识别-有线表格](https://www.modelscope.cn/models/damo/cv_dla34_table-structure-recognition_cycle-centernet)
[读光-表格结构识别-无线表格](https://www.modelscope.cn/models/damo/cv_resnet-transformer_table-structure-recognition_lore/summary)
如果效果满足，会将其整合到现有仓库下，欢迎有兴趣的小伙伴一起做。
详见 https://github.com/RapidAI/RapidStructure/issues/2

**Comment by nissansz** [comment]
![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/0b78fdeb-27bd-46a7-b81d-d927817f7762)
阿里demo结果也是一样的。你的现在的有线表格模型本身就是从阿里转换来的吧。
结果一样

**Comment by qqeip** [comment]
感觉是有个斜线

加个判断就可 
def combine_two_poly(polygons: np.ndarray, idxs: np.ndarray) -> np.ndarray:
    del_idxs, insert_boxes = [], []
    # print(idxs)
    if idxs.shape[1] == 1:
        idxs = idxs.flatten()
    else:
        idxs = idxs.squeeze(0)

**Reply by nissansz** [reply]
修改后，的确可以了，还是表格线识别不准确，导致结构错误，不知道怎么改善。
![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/a04da578-dfd3-4396-b754-29f84b0b91ea)

**Reply by nissansz** [reply]
识别结果的整合问题。
有时，用上面的有线表格模型识别，有如下问题，
![Pages from Olympos_FLSS (4)-2 - 上表](https://github.com/RapidAI/TableStructureRec/assets/41010669/9cddcc86-889f-465d-8c85-1fec2b1a99d6)

![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/ff7cd54d-52c0-4e88-bd92-92078f715550)

感觉好像是单元格polygon切割后再识别，切割有问题。

之前用paddlestruture模型，是整页文字识别，不会有这种问题，不知道有没有办法，改成之前那种整合方法。全页文字识别，然后排序填充

**Reply by nissansz** [reply]
同样，
![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/c21d2e87-9cbe-404e-9f2c-54b90f2b7b0b)

倾斜的表格，不知道要怎么改代码，让单元格对齐，正常，现在的结果如下
![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/9a160ccc-65d3-4351-bba7-ef030b2608fc)

**Reply by qqeip** [reply]
你参考 lineless main.py
    def __call__(self, content: Dict[str, Any]) -> str:
        ss = time.perf_counter()
        img = self.load_img(content)

        ocr_res, _ = self.ocr(img)

        input_info = self.preprocess(img)
        try:
            polygons, slct_logi = self.infer(input_info)
            logi_points = self.filter_logi_points(slct_logi)

            sorted_polygons = sorted_boxes(polygons)

            cell_box_map = match_ocr_cell(sorted_polygons, ocr_res)

**Reply by qqeip** [reply]
都是先识别再填充的 看代码

**Reply by nissansz** [reply]
如果我想检测图片中表的坐标，有什么onnx模型可用吗？

**Reply by nissansz** [reply]
改成ctpn检测，文字识别改善了。
![image](https://github.com/RapidAI/TableStructureRec/assets/41010669/b0be50f2-3e79-4f61-9c79-d5423ed566f1)

**Reply by nissansz** [reply]
开了一个新discussion，不知道怎么改善恢复表格结构的代码，可以发个修改后的完整代码试试？
https://github.com/RapidAI/TableStructureRec/discussions/6

**Comment by nissansz** [comment]
合合的表格识别好像都是正确的，不知道是怎么训练的。

**Reply by qqeip** [reply]
这个没关注 感觉先预处理 固定模板位置的取 参考iocr

**Reply by nissansz** [reply]
iocr是github项目？有链接吗？

**Reply by qqeip** [reply]
ocr_sdk
https://www.aias.top/products/cv/ocr_sdks/ocr_sdk.html

**Reply by qqeip** [reply]
https://github.com/mymagicpower/AIAS

**Reply by nissansz** [reply]
这个是免费的吗？没看到测试上传按钮。
http://aias.top/products/app/iocr.html

**Reply by qqeip** [reply]
看git项目

**Reply by nissansz** [reply]
git上没看到表格介绍

**Reply by qqeip** [reply]
https://github.com/mymagicpower/AIAS/issues/3
可能把iocr去掉了 你历史版本上找找

**Comment by SWHL** [comment]
@qqeip  有兴趣一起维护这个表格识别项目吗？感兴趣的话，可以加个QQ，我的是1226778264
